### PR TITLE
Treat floating-point epoch values as floating-point values

### PR DESCRIPTION
**Pull request by chansen**
This pull request fixes a bug when ->from_epoch is invoked negative fractional seconds, It also changes the way ->from_epoch handles floating-point epoch values.

### CODE REVIEW

**Comment by autarch** [comment]
Travis is failing because the code doesn't follow my style. If you run perltidy using the config in the repo, that should fix it.

File: lib/DateTime.pm [hunk 1]
```diff
@@ -2506,11 +2513,8 @@ This class method can be used to construct a new DateTime object from
 an epoch time instead of components. Just as with the C<new()>
 method, it accepts "time_zone", "locale", and "formatter" parameters.
 
-If the epoch value is not an integer, the part after the decimal will
-be converted to nanoseconds. This is done in order to be compatible
-with C<Time::HiRes>. If the floating portion extends past 9 decimal
-places, it will be truncated to nine, so that 1.1234567891 will become
-1 second and 123,456,789 nanoseconds.
+If the epoch value is a floating-point value, it will be rounded to
+nearest microsecond.
```
Comment (autarch) [inline lib/DateTime.pm]: Shouldn't this be nanosecond?

File: t/04epoch.t [hunk 2]
```diff
@@ -116,8 +116,19 @@ use DateTime;
 }
 
 {
+    my $dt = DateTime->from_epoch( epoch => -0.5 );
+    is(
+        $dt->nanosecond, 500_000_000,
+        'nanosecond should be 500,000,000 with 0.5 as epoch'
```
Comment (autarch) [inline t/04epoch.t]: This should say -0.5

File: lib/DateTime.pm (hunk 1 above)
Comment (autarch) [inline lib/DateTime.pm]: Actually, I see that you *are* rounding to the millisecond, but why? Wouldn't it be just as easy to round to the nanosecond? What am I missing?

Comment (chansen) [inline lib/DateTime.pm]: Perl is typically compiled with NV as a double. A double with a significand precision of 53 bits can only represent a nanosecond epoch without loss of precision if the duration from zero epoch is less than ≈ ±104 days. With microseconds the duration is ±104000 days, which is ~ ±285 years.

The code is rounded to nearest **microsecond**, not millisecond!

**Comment by autarch** [comment]
Thanks. I merged this from the CLI.

**Pull request merged**